cli/scorecard/score.go

```diff
@@ -148,19 +150,67 @@ func (inventory *InventoryConfig) Score(config *ScoringConfig) error {
 	err = config.attachViolations(auditResult)
 
 	if len(auditResult.Violations) > 0 {
-		fmt.Printf("\n\n%v total issues found\n", len(auditResult.Violations))
-		for _, category := range config.categories {
-			fmt.Printf("\n\n%v: %v issues found\n", category.Name, category.Count())
-			fmt.Printf("----------\n")
-			for _, cv := range category.constraints {
-				fmt.Printf("%v: %v issues\n", cv.GetName(), cv.Count())
-				for _, v := range cv.Violations {
-					fmt.Printf("- %v\n\n",
-						v.Message,
-					)
-					Log.Debug("Violation metadata", "metadata", v.GetMetadata())
+		content := ""
+		switch outputFormat {
+		case "json":
+			type violationOutput struct {
```

Comment:
Let's not define a type inline here. Instead maybe we can define a new `RichViolation` struct? We should also have a constructor for said struct which can create it from a violation + category.

Reply:
addressed in 372a085